kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java

```diff
@@ -204,19 +257,31 @@ protected void doGet(final CloudletTopic reqTopic, final KuraRequestPayload reqP
             this.findAssets();
             if (reqTopic.getResources().length == 1) {
                 this.getAllAssets(respPayload);
+                return;
             }
             // Checks if the name of the asset is provided
-            if (reqTopic.getResources().length == 2) {
-                final String assetPid = reqTopic.getResources()[1];
-                this.getAllChannelsByAssetPid(respPayload, assetPid);
+            if (reqTopic.getResources().length != 2) {
```

Comment:
You can extract `reqTopic.getResources()` as a variable so that you don't need to call it again thereafter.